File: src/main/java/com/icube/sim/tichu/games/tichu/dtos/TrickDto.java
package com.icube.sim.tichu.games.tichu.dtos;

import com.icube.sim.tichu.games.tichu.tricks.TrickType;
import org.jspecify.annotations.Nullable;

import java.util.List;

public record TrickDto(
        int playerIndex,
        TrickType type,
        List<CardDto> cards,
        @Nullable Float rank,
        @Nullable Integer minRank,
        @Nullable Integer maxRank,
        @Nullable Float phoenixRank
) {
}

File: frontend/src/games/tichu/dtos/TrickDto.ts
import { PlayerIndex } from "@/games/tichu/types.ts";
import { TrickType } from "@/games/tichu/domain/Trick.ts";
import { CardDto } from "@/games/tichu/dtos/CardDto.ts";

export interface TrickDto {
  playerIndex: PlayerIndex
  type: TrickType
  cards: CardDto[]
  rank: number | null
  minRank: number | null
  maxRank: number | null
  phoenixRank: number | null
}
